### PR TITLE
Ohai::Application loads configuration file

### PR DESCRIPTION
**Pull request by mcquin**
Adds support for loading a configuration file when `ohai` is run as an application, as specified in [rfc053](https://github.com/chef/chef-rfc/blob/master/rfc053-ohai-config.md#configuration-files)

@chef/client-core @chef/client-maintainers

### CODE REVIEW

**Comment by mcquin** [comment]
ping @chef/client-core

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,11 @@ source "https://rubygems.org"
 
 gemspec
 
+# Bundle chef-config gem here until there is a release which includes
+# workstation_config_loader.rb (12.5?)
+gem 'chef-config', :git => 'https://github.com/chef/chef.git',
```
Comment (jaym) [inline Gemfile]: i think if you merge this, our builds will break. Just keep an eye out for that

Comment (mcquin) [inline Gemfile]: /awesome/

**Comment by jaym** [comment]
:+1:

File: ohai.gemspec [hunk 2]
```diff
@@ -25,7 +25,6 @@ Gem::Specification.new do |s|
   s.add_dependency "ipaddress"
   s.add_dependency "wmi-lite", "~> 1.0"
   s.add_dependency "ffi", "~> 1.9"
-  s.add_dependency "chef-config", "~> 12.4"
```
Comment (thommay) [inline ohai.gemspec]: we still need this dependency don't we?

**Comment by ksubrama** [comment]
Hey I see you've refactored the parts where the log location is set.  Is this to help with https://github.com/chef/chef/pull/3689 ?  Or is that still a separate issue?

**Comment by mcquin** [comment]
@ksubrama 3869 is still a separate issue. Configuring `Ohai::Log` in `Ohai::System` with config settings is functionally the same as before. I pulled that bit out of `Ohai::Application` so we don't do it twice (not that it really matters).

**Comment by ksubrama** [comment]
:+1:

File: ohai.gemspec (hunk 2 above)
Comment (mcquin) [inline ohai.gemspec]: We will. Once there is a release of `chef-config` with [this change](https://github.com/chef/chef/commit/116e5219c717614658ecf2d7bf7f6b3420a8f6fe) we can re-add and update the version dep.

Comment (mcquin) [inline ohai.gemspec]: It's currently specified in the Gemfile.

**Comment by danielsdeleo** [comment]
:+1: though looks like we have to ship Chef Client first before we can merge for reals.

**Comment by mcquin** [comment]
Closing, because I want to merge this branch into master and not mcquin/ohai-config/deprecate-config (which is in master now).

**Pull request closed without merging**